frontend/src/LogIn/LogIn.jsx

```diff
@@ -0,0 +1,101 @@
+import './LogIn.css'
+import React, { useState, useContext, createContext} from "react";
+import { Link } from 'react-router-dom'
+import {useNavigate} from "react-router-dom"
+// import  {UserContext}  from '../UserContext'
+
+
+function LogIn(){
+    const [email, setEmail] = useState('');
+    const [password, setPassword] = useState('');
+    const [error, setError] = useState(null);
+    const [loading, setLoading] = useState(false);
+    const navigate = useNavigate()
+    // const {updateUser} = useContext(UserContext);
+
+
+
+
+    async function handleLogIn(event){
+        event.preventDefault();
+        if(!email || !password){
+            setError("Fill out both fields")
+            return;
+        }
+
+        try{
+            setLoading(true);
+            const response = await fetch("http://127.0.0.1:2500/logIn",{
+                method:"POST",
+                headers:{
+                    "Content-Type": "application/json",
+                },
+                body:JSON.stringify({
+                    email,
+                    password,
+                }),
+                credentials: 'include'
+            });
+            console.log(response )
+            if(response.ok){
+                const data = await response.json();
+                const loggedInUser = data.user;
+                // updateUser(loggedInUser);
+                navigate("/HomePage")
+            }else{
+                setError('Login Failed')
```

Comment:
this string is used in multiple places. Can you create a separate file for strings that used in multiple places and move them there?